apps/web/src/components/activity/ExecutionDetailsModal.tsx

```diff
@@ -20,11 +23,20 @@ export function ExecutionDetailsModal({
   onViewDigestExecution?: (digestNotificationId: string) => void;
 }) {
   const theme = useMantineTheme();
+  const [shouldRefetch, setShouldRefetch] = useState(true);
```

Comment:
```suggestion
  const shouldRefetch = status && status !== ExecutionDetailsStatusEnum.PENDING;
```

I think we can actually remove this state as I think it's not really needed and makes the code a bit easier to read :)

Reply:
My problem with that was that I don't actually have the status until after the first run of the query. As I need the response to get the status

Reply:
Oh got ya, so you need the default here to initially fetch it. 🤞